ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -138,18 +131,63 @@
     this.notifier = notifier;
     this.scheduler = scheduler;
     this.config = config;
+    this.accountInfoMapRef = new AtomicReference<>(new AccountInfoMap(accountServiceMetrics));
+    this.backup = new LocalBackup(this.accountServiceMetrics, config);
 
-    backupDirPath = config.backupDir.isEmpty() ? null : Files.createDirectories(Paths.get(config.backupDir));
     if (notifier != null) {
       notifier.subscribe(ACCOUNT_METADATA_CHANGE_TOPIC, changeTopicListener);
     } else {
       logger.warn("Notifier is null. Account updates cannot be notified to other entities. Local account cache may not "
           + "be in sync with remote account data.");
       accountServiceMetrics.nullNotifierCount.inc();
     }
+    if (config.useNewZNodePath) {
+      accountMetadataStore = new RouterStore(this.accountServiceMetrics, backup, helixStore, router);
```

Comment:
I understand this is first PR to improve account metadata and I believe `setupRouter` will be invoked in future PR, however, here `router` might not be initialized, can we do some proactive check in `RouterStore`?
Eventually, after transition is done, we should explicitly call `setupRouter` in `RestServer.java`

Reply:
good idea, updated